src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -50,28 +60,177 @@
   /**
    * Get the historical price for each symbol passed in for the date passed in.
    *
-   * @param symbol list of symbols to get historical price for.
+   * @param symbol Stock symbol to get historical price for.
    * @param range the range of time  (ex. "5m", "ytd" ) to get previous data (Optional).
    * @param date the date from which we would want to get the previous data from (Optional).
-   * Note: If neither optional parameter is used, the system will default to range = 1m.
-   * - default behavior of Iex Cloud API
-   * Note: If both optional parameters are provided, the system will default to the date.
-   * - can be changed later if needed: IexCloud will not allow both options in call to client
-   * @return a IexHistoricalPrices objects for the given symbols.
+   *      Note: If neither optional parameter is used, the system will default to range = 1m.
+   *      - default behavior of Iex Cloud API
+   *      Note: If both optional parameters are provided, the system will default to the date.
+   *      - can be changed later if needed: IexCloud will not allow both options in call to client
+   * @return a list of IexHistoricalPrices objects for the given symbols.
    */
   public List<IexHistoricalPrices> getHistoricalPricesForSymbol(
-      final String symbol, String range, String date) {
-    if (symbol==null) {
+      final String symbol, final String range, final String date) {
+
+    if (symbol == null) {
       throw new IllegalArgumentException("No Stock Symbol Provided.");
-    } else if (range==null && date==null) {
-      String rangeVal = "1m";
-      return iexClient.getHistoricalPricesForSymbolByRange(symbol, rangeVal);
-    } else if (date==null) {
-      return iexClient.getHistoricalPricesForSymbolByRange(symbol, range);
+    } else if (!isSymbolInDatabase(symbol)) {
```

Comment:
Why is this check here? The `getHistoricalPriceBySymbolAndDate` seems to be in charge of deciding whether to go to IEX or not, shouldn't this DB related check be in that method?

Reply:
I included this to be a "shortcut" for the system if we don't have any stock of that type in the database at all. Essentially, this function will allow for a "range" call to Iex, since it doesn't need to parse through each date and check whether it is in the database. This also allows for 1 range batch call into Iex, versus the smaller 1-date-at-a-time calls. I can certainly eliminate this if you want, but I think it would save sufficient amount of time to be worth running this check in each call.